File: app/model/Encounter.js
// app/model/Encounter.js
import { Model } from '@nozbe/watermelondb';
import { field, text, date, relation, json } from '@nozbe/watermelondb/decorators';

export default class Encounter extends Model {
  static table = 'encounters';
  static associations = {
    patients: { type: 'belongs_to', foreignKey: 'patient_id' },
    diagnostics: { type: 'has_many', foreignKey: 'encounter_id' },
  };

  @relation('patients', 'patient_id') patient;
  @text('soap_note') soapNote;
  @text('voice_transcript') voiceTranscript;
  @field('audio_r2_key') audioR2Key;
  @field('sp_o2') spO2;
  @field('heart_rate') heartRate;
  @field('systolic') systolic;
  @field('age') age;
  @field('pregnant') pregnant;
  @field('triage_score') triageScore;
  @json('triage_rationale', (json) => json) triageRationale;
  @text('diagnosis') diagnosis;
  @text('treatment_plan') treatmentPlan;
  @text('medicines') medicines;
  @text('safety_notes') safetyNotes;
  @field('confidence') confidence;
  @text('triage') triage;
  @field('sync_status') syncStatus;
  @date('created_at') createdAt;
  @date('synced_at') syncedAt;
}

File: src/main/java/com/carenode/entity/Encounter.java
package com.carenode.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "encounters")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Encounter {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private UUID id;

    @ManyToOne
    @JoinColumn(name = "patient_id")
    private Patient patient;

    @Column(columnDefinition = "TEXT")
    private String soapNote;

    @Column(columnDefinition = "TEXT")
    private String voiceTranscript;

    @Column(name = "audio_r2_key")
    private String audioR2Key;

    @Column(name = "spO2")
    private Integer spo2;
    private Integer heartRate;
    private Integer systolic;
    private Integer age;
    private Boolean pregnant;

    private String triageScore;

    @Column(columnDefinition = "jsonb")
    private String triageRationale;

    @Column(columnDefinition = "TEXT")
    private String diagnosis;

    @Column(columnDefinition = "TEXT")
    private String treatmentPlan;

    @Column(columnDefinition = "TEXT")
    private String medicines;

    @Column(columnDefinition = "TEXT")
    private String safetyNotes;

    private Double confidence;

    private String triage;

    @Enumerated(EnumType.STRING)
    private SyncStatus syncStatus = SyncStatus.PENDING;

    @PrePersist
    private void onCreate() {
        createdAt = LocalDateTime.now();
    }

    private LocalDateTime createdAt;

    private LocalDateTime syncedAt;

    public enum SyncStatus {
        PENDING, SYNCED, CONFLICT
    }
}